crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2909,7 +2910,7 @@ mod tests {
 
         // TODO: update this once `infer_ellipsis_literal_expression` correctly
         // infers `types.EllipsisType`.
-        assert_public_ty(&db, "src/a.py", "x", "Unknown | Literal[EllipsisType]");
+        assert_public_ty(&db, "src/a.py", "x", "Unbound");
```

Comment:
Arguably this is actually an improvement, since at least I understand how we arrive at this conclusion now 😆 `Literal[EllipsisType]` is pretty baffling; I don't understand why we think `Ellipsis` is equal to the `EllipsisType` class object

Reply:
It's because of the wrong code in `infer_annotated_assignment` that I removed in this PR, where we inferred the annotation as a value expression and assign that type to the name.

Reply:
Ah I see... thanks!